src/App.tsx

```diff
@@ -1,24 +1,200 @@
+    try {
+      setLoadingIds(state => [...state, todoId]);
+      await deleteTodo(todoId);
+      setTodos(todos.filter(todo => todo.id !== todoId));
+    } catch {
+      setErrorOfUpdate('Unable to delete a todo');
+      setTimeout(() => {
+        setErrorOfUpdate('');
+      }, 3000);
+    } finally {
+      setLoadingIds([]);
+    }
+  };
+
+  const clearCompletedTodos = () => {
+    const completedTodos = todos.filter(todo => todo.completed);
+
+    completedTodos.forEach(todo => {
+      deleteTodo(todo.id)
+        .then(() => {
+          setTodos(todos.filter(todoItem => !todoItem.completed));
+        })
+        .catch(() => {
+          setErrorOfUpdate('Unable to delete todos');
+        });
+    });
+  };
+
+  const handleSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+
+    addSomeTodo(textOfInput);
+    setTextOfInput('');
+  };
+
+  const handleChange = (event: React.KeyboardEvent<HTMLInputElement>) => {
+    setTextOfInput(event.target.value);
+  };
+
+  useEffect(() => {
+    setErrorOfUpdate('');
+    getTodosFromServer();
+    setTimeout(() => setErrorOfUpdate(''), 3000);
+  }, []);
+
+  const visibleTodos = getFilteredTodos(filterOption, todos);
+
   if (!USER_ID) {
     return <UserWarning />;
   }
 
   return (
-    <section className="section container">
-      <p className="title is-4">
-        Copy all you need from the prev task:
-        <br />
-        <a href="https://github.com/mate-academy/react_todo-app-loading-todos#react-todo-app-load-todos">React Todo App - Load Todos</a>
-      </p>
-
-      <p className="subtitle">Styles are already copied</p>
-    </section>
+    <div className="todoapp">
+      <h1 className="todoapp__title">todos</h1>
+
+      <div className="todoapp__content">
+        <header className="todoapp__header">
+          <button type="button" className="todoapp__toggle-all active" />
+          <form onSubmit={handleSubmit}>
+            <input
+              type="text"
+              className="todoapp__new-todo"
+              placeholder="What needs to be done?"
+              value={textOfInput}
+              onChange={handleChange}
+              disabled={inputStatus}
+            />
+          </form>
+        </header>
+
+        <section className="todoapp__main">
+          {isTodosLoading
+            ? <Loader />
+            : (
+              <TodoList
+                todos={visibleTodos}
+                tempTodo={tempTodo}
+                onDelete={deleteSomeTodo}
+                loadingIds={loadingIds}
+              />
+            )}
+        </section>
+
+        {todos && (
+          <TodoFilter
+            filterOption={filterOption}
+            setFilterOption={setFilterOption}
+            numberOfActiveTodos={numberOfActiveTodos}
+            numberOfCompletedTodos={numberOfCompletedTodos}
+            onClearCompleted={clearCompletedTodos}
+          />
+        )}
+      </div>
+      <div
```

Comment:
I'd recommend you create a separate component for the error message and implement there removing it from the page after 3 sec (using `useEffect`)

Reply:
done